src/audio/kpb.c

```diff
@@ -1327,39 +1386,16 @@ static enum task_state kpb_draining_task(void *arg)
 static void kpb_drain_samples(void *source, struct audio_stream *sink,
 			      size_t size, size_t sample_width)
 {
-#if CONFIG_FORMAT_S16LE || CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE
-	void *dst;
-	void *src = source;
-	size_t i;
-	size_t j = 0;
-	size_t channel;
-	size_t frames = KPB_BYTES_TO_FRAMES(size, sample_width);
-#endif
-
 	switch (sample_width) {
 #if CONFIG_FORMAT_S16LE
 	case 16:
-		for (i = 0; i < frames; i++) {
-			for (channel = 0; channel < KPB_NUM_OF_CHANNELS; channel++) {
-				dst = audio_stream_write_frag_s16(sink, j);
-				*((int16_t *)dst) = *((int16_t *)src);
-				src = ((int16_t *)src) + 1;
-				j++;
-			}
-		}
+		drain_s16(source, sink, size);
 		break;
 #endif /* CONFIG_FORMAT_S16LE */
 #if CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE
-	case 24:
+	case 24: /* FALLTHROUGH */
```

Comment:
we use `COMPILER_FALLTHROUGH` in cases like
```
switch (x) {
case A:
	do_stuff();
	COMPILER_FALLTHROUGH;
case B:
```
but where multiple cases do exactly the same like
```
switch (x) {
case A:
case B:
	do_stuff();
```
in many locations we have no comments and I don't think any static analysers or compiler versions ever complained. Your comment doesn't hurt anyone of course, and if it helps - no problem at all, but just I wouldn't begin adding these comments to all such cases... :-)

Reply:
Thanks for confirm, I'll remove. I saw these elsewhere in the module and followed.

Reply:
fixed